arangod/IResearch/IResearchFeature.cpp

```diff
@@ -277,6 +454,44 @@ void registerRecoveryHelper() {
   }
 }
 
+void registerUpgradeTasks() {
+  auto* upgrade =
+      arangodb::application_features::ApplicationServer::lookupFeature<arangodb::UpgradeFeature>(
+          "Upgrade");
+
+  if (!upgrade) {
+    return;  // nothing to register with (OK if no tasks actually need to be applied)
+  }
+
+  // move IResearch data-store from IResearchView to IResearchLink
+  {
+    arangodb::methods::Upgrade::Task task;
+
+    task.name = "IResearhView version 0->1";
+    task.description =
+        "move IResearch data-store from IResearchView to IResearchLink";
+    task.systemFlag = arangodb::methods::Upgrade::Flags::DATABASE_ALL;
+    task.clusterFlags = arangodb::methods::Upgrade::Flags::CLUSTER_COORDINATOR_GLOBAL  // any 1 single coordinator
```

Comment:
I don't get it, is it necessary to really drop the view on the coordinator ?

Reply:
you mean Upgrade::Flags::CLUSTER_COORDINATOR_GLOBAL?
this is to ensure that:
* the view definition in the agency is updated
* the links get forcefully recreated on all the dbservers